bundle/config/mutator/translate_paths.go

```diff
@@ -218,61 +224,65 @@ to contain one of the following file extensions: [%s]`, literal, strings.Join(ex
 	}
 
 	// Upon import, notebooks are stripped of their extension.
-	localRelPathNoExt := strings.TrimSuffix(localRelPath, filepath.Ext(localRelPath))
-	return path.Join(t.remoteRoot, filepath.ToSlash(localRelPathNoExt)), nil
+	localRelPathNoExt := strings.TrimSuffix(localRelPath, path.Ext(localRelPath))
+	return path.Join(t.remoteRoot, localRelPathNoExt), nil
 }
 
 func (t *translateContext) translateFilePath(ctx context.Context, literal, localFullPath, localRelPath string) (string, error) {
-	nb, _, err := notebook.DetectWithFS(t.b.SyncRoot, filepath.ToSlash(localRelPath))
+	nb, _, err := notebook.DetectWithFS(t.b.SyncRoot, localRelPath)
 	if errors.Is(err, fs.ErrNotExist) {
 		return "", fmt.Errorf("file %s not found", literal)
 	}
 	if err != nil {
-		return "", fmt.Errorf("unable to determine if %s is not a notebook: %w", localFullPath, err)
+		return "", fmt.Errorf("unable to determine if %s is not a notebook: %w", filepath.FromSlash(localFullPath), err)
 	}
 	if nb {
 		return "", ErrIsNotebook{localFullPath}
 	}
-	return path.Join(t.remoteRoot, filepath.ToSlash(localRelPath)), nil
+	return path.Join(t.remoteRoot, localRelPath), nil
 }
 
 func (t *translateContext) translateDirectoryPath(ctx context.Context, literal, localFullPath, localRelPath string) (string, error) {
-	info, err := t.b.SyncRoot.Stat(filepath.ToSlash(localRelPath))
+	info, err := t.b.SyncRoot.Stat(localRelPath)
 	if err != nil {
 		return "", err
 	}
 	if !info.IsDir() {
-		return "", fmt.Errorf("%s is not a directory", localFullPath)
+		return "", fmt.Errorf("%s is not a directory", filepath.FromSlash(localFullPath))
 	}
-	return path.Join(t.remoteRoot, filepath.ToSlash(localRelPath)), nil
+	return path.Join(t.remoteRoot, localRelPath), nil
 }
 
 func (t *translateContext) translateLocalAbsoluteFilePath(ctx context.Context, literal, localFullPath, localRelPath string) (string, error) {
-	info, err := t.b.SyncRoot.Stat(filepath.ToSlash(localRelPath))
+	info, err := t.b.SyncRoot.Stat(localRelPath)
 	if errors.Is(err, fs.ErrNotExist) {
 		return "", fmt.Errorf("file %s not found", literal)
 	}
 	if err != nil {
-		return "", fmt.Errorf("unable to determine if %s is a file: %w", localFullPath, err)
+		return "", fmt.Errorf("unable to determine if %s is a file: %w", filepath.FromSlash(localFullPath), err)
 	}
 	if info.IsDir() {
 		return "", fmt.Errorf("expected %s to be a file but found a directory", literal)
 	}
-	return localFullPath, nil
+	// Absolute paths are never externalized and always intended to be used locally.
+	// Return a platform-native path.
+	return filepath.FromSlash(localFullPath), nil
```

Comment:
> // Absolute paths are never externalized and always intended to be used locally.

But the can still show up the output, thus making test output depend on the OS

Don't understand the value of calling FromSlash, should not Windows just work with forward slashes?

Reply:
I'll dig a bit more.

I recall calling `os.*` and `exec.*` functions with slash-separated paths being problematic on Windows.

Reply:
There are two places where an absolute path is used:
1. To refer to a dashboard file when passing the reference to Terraform
2. To synthesize a working directory for building an artifact

Both of these work with forward slash-separated absolute paths out of the box.

It's simpler now in that all paths translated by this mutator are forward slash-separated.